lib/statsample-glm/glm/mle/base.rb

```diff
@@ -109,11 +109,11 @@ def newton_raphson(x,y, start_values=nil)
             @iterations = i + 1
 
             h = second_derivative(x,y,parameters)
-            if h.singular?
+            if h.det() == 0
```

Comment:
Not sure if this check is necessary. Computing the determinant is very expensive. I think we can continue without it, then methods like `invert` or `solve` will complain later on anyway, if h is (nearly) singular.